lib/grok-pure.rb

```diff
@@ -144,7 +144,9 @@ def compile_captures_func(re)
                    else; ""
                  end
         name = pattern if name.nil?
-        re_match << "  block.call(#{name.inspect}, match[#{index}]#{coerce})"
+        # return the capture name, coerced value and original value to the caller
+        # we do all this coerce logic at compile time to optimize for run time
+        re_match << "  block.call(#{name.inspect}, match[#{index}]#{coerce}, match[#{index}])"
```

Comment:
Unless we decide that there's a good reason to return both the original and coerced matches, this line could be changed to:

```
re_match << "  m = match[#{index}]"
re_match << "  block.call(#{name.inspect}, (m ? m#{coerce}, m)"
```

this would also simplify https://github.com/logstash-plugins/logstash-filter-grok/pull/20: we only need to add the specs and not touch lib/logstash/filters/grok.rb